lib/util/selector-event-generator.js

```diff
@@ -0,0 +1,129 @@
+/**
+ * @fileoverview The event generator for AST nodes.
+ * @author Ilya Volodin
+ * @copyright 2015 Ilya Volodin. All rights reserved.
+ * See LICENSE file in root directory for full license.
+ */
+
+"use strict";
+
+var selectorParser = require("./selector-parser");
+
+//------------------------------------------------------------------------------
+// Private helpers
+//------------------------------------------------------------------------------
+
+/**
```

Comment:
I'd bundle anything doing the matching in with the parsing to keep related functionality together.